splash/browser_tab.py

```diff
@@ -892,6 +904,12 @@ def _send_request(self, url, callback, method='GET', body=None,
 
         request = self.request_obj(url, headers=headers, body=body)
 
+        if headers and "user-agent" in [k.lower() for k in headers.keys()]:
+            user_agent = headers.get("user-agent", headers.get("User-Agent"))
+        else:
+            user_agent = self.web_page.userAgentForUrl(to_qurl(url))
+        request.setRawHeader(b"user-agent", to_bytes(user_agent))
```

Comment:
splash go() has two flows when creating new requests. When you pass baseurl request is downloaded by splash http client. If you dont pass baseurl it is downloaded with web page. request_obj is used in both of these scenarios. But User agent is missing **only** if you pass base_url. If user-agent would be set in request_obj it would be set also for request downloads going via webpage which is not necessary. This is why it is set here.